src/librustdoc/clean/types.rs

```diff
@@ -57,20 +57,34 @@ pub(crate) type ItemIdSet = FxHashSet<ItemId>;
 #[derive(Debug, Clone, PartialEq, Eq, Hash, Copy)]
 pub(crate) enum ItemId {
     /// A "normal" item that uses a [`DefId`] for identification.
-    DefId(DefId),
+    DefId(DefId, Option<Namespace>),
```

Comment:
I'm confused by the ability to set a `None` namespace here. Could you elaborate on the semantics of the new value here?

Also, I believe some items can cause names to appear in multiple namespaces — for example, unit structs. What would the namespace value be set to in such a case?

Reply:
Initial test failures led me to believe that DefIds didn't carry namespace information, but that guess turned out incorrect, I've changed the impl now